pkg/build/admission/strategyrestrictions/admission.go

```diff
@@ -157,8 +168,22 @@ func (a *buildByStrategy) checkAccess(strategy buildapi.BuildStrategy, subjectAc
 	if err != nil {
 		return admission.NewForbidden(attr, err)
 	}
+	// If not allowed, try to check against the legacy resource
+	// FIXME: Remove this when the legacy API is deprecated
```

Comment:
That is actually a good idea. It will also allow registry to send SAR to openshift with list of images that needs to be checked instead of doing one-by-one request. @enj ?